nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/MoveAzureDataLakeStorage.java

```diff
@@ -0,0 +1,223 @@
+                    " If the original directory structure should be kept, the full directory path needs to be provided after the destination directory." +
+                    " e.g.: destdir/${azure.directory}")
+            .addValidator(new DirectoryValidator("Destination Directory"))
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+            ADLS_CREDENTIALS_SERVICE,
+            SOURCE_FILESYSTEM,
+            SOURCE_DIRECTORY,
+            DESTINATION_FILESYSTEM,
+            DESTINATION_DIRECTORY,
+            FILE,
+            CONFLICT_RESOLUTION
+    ));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final long startNanos = System.nanoTime();
+        try {
+            final String sourceFileSystem = evaluateFileSystemProperty(context, flowFile, SOURCE_FILESYSTEM);
+            final String sourceDirectory = evaluateDirectoryProperty(context, flowFile, SOURCE_DIRECTORY);
+            final String destinationFileSystem = evaluateFileSystemProperty(context, flowFile, DESTINATION_FILESYSTEM);
+            final String destinationDirectory = evaluateDirectoryProperty(context, flowFile, DESTINATION_DIRECTORY);
+            final String fileName = evaluateFileNameProperty(context, flowFile);
+
+            String destinationPath = destinationDirectory;
+            if (!destinationDirectory.isEmpty() && ! sourceDirectory.isEmpty()) {
+                destinationPath = destinationDirectory + "/";
+            }
+
+            final DataLakeServiceClient storageClient = getStorageClient(context, flowFile);
+            final DataLakeFileSystemClient sourceFileSystemClient = storageClient.getFileSystemClient(sourceFileSystem);
+            final DataLakeDirectoryClient sourceDataLakeDirectoryClient = sourceFileSystemClient.getDirectoryClient(sourceDirectory);
+            final DataLakeFileSystemClient destinationFileSystemClient = storageClient.getFileSystemClient(destinationFileSystem);
+            final DataLakeDirectoryClient destinationDirectoryClient = destinationFileSystemClient.getDirectoryClient(destinationDirectory);
+            DataLakeFileClient fileClient = sourceDataLakeDirectoryClient.getFileClient(fileName);
+            final DataLakeRequestConditions sourceConditions = new DataLakeRequestConditions();
+            final DataLakeRequestConditions destConditions = new DataLakeRequestConditions();
+            final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+
+            try {
```

Comment:
I think this nested try may be extracted to a separate method to make the code more readable.